Compiler error using example simpleAsync-esp32.ino.ino

using Arduino IDE 1.8.19 with "AI Thinker ESP32-CAM" board.


```
In file included from D:\SYS_ext\Joerg\Documents\Arduino\libraries\espMqttClient-1.4.2\src/MqttClient.h:22:0,
                 from D:\SYS_ext\Joerg\Documents\Arduino\libraries\espMqttClient-1.4.2\src/MqttClientSetup.h:14,
                 from D:\SYS_ext\Joerg\Documents\Arduino\libraries\espMqttClient-1.4.2\src/espMqttClientAsync.h:18,
                 from D:\SYS_ext\Joerg\Documents\Arduino\simpleAsync-esp32.ino\simpleAsync-esp32.ino.ino:3:
D:\SYS_ext\Joerg\Documents\Arduino\libraries\espMqttClient-1.4.2\src/Packets/Packet.h: In constructor 'espMqttClientInternals::Packet::Packet(espMqttClientTypes::Error&, uint16_t, const char*, uint8_t, const char*, uint8_t, Args&& ...)':
D:\SYS_ext\Joerg\Documents\Arduino\libraries\espMqttClient-1.4.2\src/Packets/Packet.h:100:43: error: expected ',' before ')' token
     static_assert(sizeof...(Args) % 2 == 0);
                                           ^
D:\SYS_ext\Joerg\Documents\Arduino\libraries\espMqttClient-1.4.2\src/Packets/Packet.h:100:43: error: expected string-literal before ')' token

```

Thanks

Joerg

--- a/scripts/CI/platformio_esp32.ini
+++ b/scripts/CI/platformio_esp32.ini
@@ -17,3 +17,6 @@ build_flags =
   -Wall
   -Wextra
   -Werror
+  -D DEBUG_ESP_MQTT_CLIENT=1
+  -D CORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_VERBOSE
+build_type = debug

--- a/scripts/CI/platformio_esp8266.ini
+++ b/scripts/CI/platformio_esp8266.ini
@@ -17,3 +17,6 @@ build_flags =
   -Wall
   -Wextra
   -Werror
+  -D DEBUG_ESP_MQTT_CLIENT=1
+  -D DEBUG_ESP_PORT=Serial
+build_type = debug

--- a/scripts/CI/platformio_esp8266.ini
+++ b/scripts/CI/platformio_esp8266.ini
@@ -17,6 +17,4 @@ build_flags =
   -Wall
   -Wextra
   -Werror
-  -D DEBUG_ESP_MQTT_CLIENT=1
-  -D DEBUG_ESP_PORT=Serial
 build_type = debug

--- a/src/Logging.h
+++ b/src/Logging.h
@@ -14,9 +14,9 @@ the LICENSE file.
   #include "freertos/task.h"
   #if defined(DEBUG_ESP_MQTT_CLIENT)
   // Logging is en/disabled by Arduino framework macros
-    #define emc_log_i(...) vPortEnterCritical(); log_i(__VA_ARGS__); vPortExitCritical()
-    #define emc_log_e(...) vPortEnterCritical(); log_e(__VA_ARGS__); vPortExitCritical()
-    #define emc_log_w(...) vPortEnterCritical(); log_w(__VA_ARGS__); vPortExitCritical()
+    #define emc_log_i(...) log_i(__VA_ARGS__)
+    #define emc_log_e(...) log_e(__VA_ARGS__)
+    #define emc_log_w(...) log_w(__VA_ARGS__)
   #else
   // Logging is disabled
     #define emc_log_i(...)

--- a/src/Packets/Packet.h
+++ b/src/Packets/Packet.h
@@ -97,7 +97,7 @@ class Packet {
   , _payloadStartIndex(0)
   , _payloadEndIndex(0)
   , _getPayload(nullptr) {
-    static_assert(sizeof...(Args) % 2 == 0);
+    static_assert(sizeof...(Args) % 2 == 0, "Subscribe should be in topic/qos pairs");
     size_t numberTopics = 2 + (sizeof...(Args) / 2);
     SubscribeItem list[numberTopics] = {topic1, qos1, topic2, qos2, args...};
     _createSubscribe(error, list, numberTopics);

